openstack/imageservice/v2/images/requests.go

```diff
@@ -0,0 +1,301 @@
+	return res
+}
+
+// CreateOptsBuilder describes struct types that can be accepted by the Create call.
+// The CreateOpts struct in this package does.
+type CreateOptsBuilder interface {
+	// Returns value that can be passed to json.Marshal
+	ToImageCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts implements CreateOptsBuilder
+type CreateOpts struct {
+	// Name [required] is the name of the new image.
+	Name *string
+
+	// Id [optional] is the the image ID.
+	ID *string
+
+	// Visibility [optional] defines who can see/use the image.
+	Visibility *ImageVisibility
+
+	// Tags [optional] is a set of image tags.
+	Tags []string
+
+	// ContainerFormat [optional] is the format of the
+	// container. Valid values are ami, ari, aki, bare, and ovf.
+	ContainerFormat *string
+
+	// DiskFormat [optional] is the format of the disk. If set,
+	// valid values are ami, ari, aki, vhd, vmdk, raw, qcow2, vdi,
+	// and iso.
+	DiskFormat *string
+
+	// MinDiskGigabytes [optional] is the amount of disk space in
+	// GB that is required to boot the image.
+	MinDiskGigabytes *int
+
+	// MinRAMMegabytes [optional] is the amount of RAM in MB that
+	// is required to boot the image.
+	MinRAMMegabytes *int
+
+	// protected [optional] is whether the image is not deletable.
+	Protected *bool
+
+	// properties [optional] is a set of properties, if any, that
+	// are associated with the image.
+	Properties map[string]string
+}
+
+// ToImageCreateMap assembles a request body based on the contents of
+// a CreateOpts.
+func (opts CreateOpts) ToImageCreateMap() (map[string]interface{}, error) {
+	body := map[string]interface{}{}
+	if opts.Name == nil {
+		return body, fmt.Errorf("'Name' field is requered, but is not set (was: %v)'", opts.Name)
+	}
+
+	body["name"] = opts.Name
+	if opts.ID != nil {
+		body["id"] = opts.ID
+	}
+	if opts.Visibility != nil {
+		body["visibility"] = opts.Visibility
+	}
+	if opts.Tags != nil {
+		body["tags"] = opts.Tags
+	}
+	if opts.ContainerFormat != nil {
+		body["container_format"] = opts.ContainerFormat
+	}
+	if opts.DiskFormat != nil {
+		body["disk_format"] = opts.DiskFormat
+	}
+	if opts.MinDiskGigabytes != nil {
+		body["min_disk"] = opts.MinDiskGigabytes
+	}
+	if opts.MinRAMMegabytes != nil {
+		body["min_ram"] = opts.MinRAMMegabytes
+
+	}
+	if opts.Protected != nil {
+		body["protected"] = opts.Protected
+	}
+	if opts.Properties != nil {
+		body["properties"] = opts.Properties
+	}
+	return body, nil
+}
+
+// Delete implements image delete request
+func Delete(client *gophercloud.ServiceClient, id string) DeleteResult {
+	var res DeleteResult
+	_, res.Err = client.Delete(deleteURL(client, id), &gophercloud.RequestOpts{
+		OkCodes: []int{204},
+	})
+	return res
+}
+
+// Get implements image get request
+func Get(client *gophercloud.ServiceClient, id string) GetResult {
+	var res GetResult
+	client.Get(getURL(client, id), &res.Body, nil)
```

Comment:
Let's keep this consistent with the other requests:
`_, res.Err = client.Get(getURL(client, id), &res.Body, nil)`